src/Data/Functor/Foldable.hs

```diff
@@ -849,3 +1114,12 @@ instance (GCoerce f g, GCoerce f' g') => GCoerce (f :*: f') (g :*: g') where
 instance (GCoerce f g, GCoerce f' g') => GCoerce (f :+: f') (g :+: g') where
     gcoerce (L1 x) = L1 (gcoerce x)
     gcoerce (R1 x) = R1 (gcoerce x)
+
+-- $intro
+-- TODO
```

Comment:
Here I want to provide a little intro to what this library is about, and the fundamental concepts users should know about. I'll also provide references to good literature for further reading.